### PR TITLE
Artifact Publishing, Default Property Value Change, Unit Tests & Documentation Updates

### PR DESCRIPTION
**Pull request by dmccoystephenson**
## Artifact Publishing
Added a GitHub action workflow to publish a Java artifact to the repository upon release creation.

### Problem
The enable.auto.commit property defaults to 'false,' but based on [issue #22](https://github.com/usdot-jpo-ode/jpo-s3-deposit/issues/22), 'true' is a more appropriate default.

### Solution
The default for enable.auto.commit is now 'true' when the environment variable is not set.

## Unit Tests Added
This PR introduces unit tests for S3-depositor methods, along with refactoring to support testability. All changes were tested and validated using Maven.

### Problem
The README was disorganized and contained inaccuracies.

### Solution
The README has been updated for better clarity and accuracy.

### CODE REVIEW

File: src/main/java/us/dot/its/jpo/ode/aws/depositor/AwsDepositor.java [hunk 1]
```diff
@@ -435,7 +404,9 @@ private JSONObject generateAWSProfile() throws IOException {
 		return jsonResult;
 	}
 
-	private static String getEnvironmentVariable(String variableName, String defaultValue) {
+	static String getEnvironmentVariable(String variableName, String defaultValue) {
+		// get all environment variables
+		Map<String, String> env = System.getenv();
```
Comment (dan-du-car) [inline src/main/java/us/dot/its/jpo/ode/aws/depositor/AwsDepositor.java]: Is this env used?

Comment (dmccoystephenson) [reply src/main/java/us/dot/its/jpo/ode/aws/depositor/AwsDepositor.java]: This variable was not used and has been removed.

**Review by dan-du-car — APPROVED** [review]

File: .github/workflows/artifact-publish.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+name: Publish Java Package
```
Comment (SaikrishnaBairamoni) [inline .github/workflows/artifact-publish.yml:1]: What will be the approx. artifact size and where we are using this ? As our ODE GitHub org is on free plan we have artifact storage limit of 500mb.

Comment (dmccoystephenson) [reply .github/workflows/artifact-publish.yml:1]: The project's JAR file is about 14 KB in size. It is not currently being used, but it is for potential future use.

Comment (SaikrishnaBairamoni) [reply .github/workflows/artifact-publish.yml:1]: Same comment from jpo-security-svcs seems like for Maven packages and other packages like npm, docker, etc storage limit will be not applied for public repo's. so I think we should be good to use artifacts. However for any private repo's this limits will apply. We have a backlog item for supporting GitHub artifacts setup in future release. Thanks!

**Review by SaikrishnaBairamoni — APPROVED** [review]

**Pull request merged**